src/utils/postgrest-data-provider.js

```diff
@@ -0,0 +1,308 @@
+import { queryParameters } from "ra-core/lib/util/fetch";
+import HttpError from "ra-core/lib/util/HttpError";
+import { GET_LIST, GET_ONE, GET_MANY, GET_MANY_REFERENCE, CREATE, UPDATE, DELETE, DELETE_MANY } from "react-admin";
+import { AUTH_LOGIN, AUTH_LOGOUT, AUTH_ERROR } from "react-admin";
+import json2ParseBigint from "./json_parse_bigint";
+
+const localStorageKey = "__hubs_admin_token";
+
+// Custom fetchJson routing to ensure bigint precision
+const fetchJson = (url, options) => {
+  const requestHeaders =
+    options.headers ||
+    new Headers({
+      Accept: "application/json"
+    });
+  if (!requestHeaders.has("Content-Type") && !(options && options.body && options.body instanceof FormData)) {
+    requestHeaders.set("Content-Type", "application/json");
+  }
+  if (options.user && options.user.authenticated && options.user.token) {
+    requestHeaders.set("Authorization", options.user.token);
+  }
+
+  return fetch(url, { ...options, headers: requestHeaders })
+    .then(response =>
+      response.text().then(text => ({
+        status: response.status,
+        statusText: response.statusText,
+        headers: response.headers,
+        body: text
+      }))
+    )
+    .then(({ status, statusText, headers, body }) => {
+      let json;
+      try {
+        json = json2ParseBigint(body);
+      } catch (e) {
+        // not json, no big deal
```

Comment:
Might be useful for debugging if we logged a warning here at least. Also maybe `reject()` here as well.